system/mp/mpcomm.cpp

```diff
@@ -463,9 +537,16 @@ class CMPConnectThread: public Thread
     void startPort(unsigned short port);
     void stop()
     {
-        if (running) {
+        if (running)
+        {
             running = false;
             listensock->cancel_accept();
+            if (listen)
+            {
+                if (!threadPool->joinAll(true, 1000*60*5))
```

Comment:
Worth commenting that stop the listening threads first in case any get added to the stop processor?